src/components/activity/ActivityDirectivesTable.svelte

```diff
@@ -171,6 +192,110 @@
     }
   }
 
+  async function updateActivities(updatedActivities: ActivityDirective[] | null) {
+    if (plan != null && Array.isArray(updatedActivities)) {
+      for (const activity of updatedActivities) {
+        const activityType = findTypes(activity.type, get(planModelActivityTypes) ?? []);
+        await effects.updateActivityDirective(
+          plan,
+          activity.id,
+          { start_offset: activity.start_offset },
+          activityType || null,
+          user && 'activeRole' in user ? (user as User) : null,
+        );
+      }
+    }
+  }
+
+  function bulkPackLeftItems() {
+    const selectedIdSet = new Set(bulkSelectedActivityDirectiveIds);
+    const selectedActivityDirectives = activityDirectives?.filter(ad => selectedIdSet.has(ad.id)) ?? [];
+
+    if (selectedActivityDirectives.length) {
+      packLeftActivityDirectives(selectedActivityDirectives);
+    }
+  }
+
+  function bulkPackRightItems() {
+    const selectedIdSet = new Set(bulkSelectedActivityDirectiveIds);
+    const selectedActivityDirectives = activityDirectives?.filter(ad => selectedIdSet.has(ad.id)) ?? [];
+
+    if (selectedActivityDirectives.length) {
+      packRightActivityDirectives(selectedActivityDirectives);
+    }
+  }
+
+  function displayPackItemsWithOffset() {
+    showPackOffsetDialog = true;
+  }
+
+  function bulkPackItemsWithOffset(event: CustomEvent<{ direction: string; gapOffset: string }>) {
+    showPackOffsetDialog = false;
+    const selectedIdSet = new Set(bulkSelectedActivityDirectiveIds);
+    const selectedActivityDirectives = activityDirectives?.filter(ad => selectedIdSet.has(ad.id)) ?? [];
+
+    const { direction, gapOffset } = event.detail;
+    if (selectedActivityDirectives.length) {
+      packActivityDirectivesOffset(selectedActivityDirectives, gapOffset, direction);
+    }
+  }
+
+  async function packLeftActivityDirectives(activities: ActivityDirective[]) {
+    if (plan !== null) {
+      const updatedActivities = packActivityDirectivesInPlan(
+        plan,
+        activities,
+        'LEFT',
+        0,
+        get(activityDirectivesDB) ?? [],
```

Comment:
Our coding pattern is to not use the `get` function to access the values of stores, but to use the `$` instead. (e.g. `$activityDirectivesDB`)